codec/src/main/java/io/netty/handler/codec/compression/Snappy.java

```diff
@@ -39,10 +48,54 @@ public final class Snappy {
     private static final int COPY_2_BYTE_OFFSET = 2;
     private static final int COPY_4_BYTE_OFFSET = 3;
 
+    // Hash table used to compress, shared between subsequent call to .encode()
+    private static final FastThreadLocal<short[]> HASH_TABLE = new FastThreadLocal<short[]>();
+
+    private static final HashType DEFAULT_HASH_TABLE_TYPE;
+
+    static {
+        String hashTableTypeKey = "io.netty.handler.codec.compression.snappy.hashTableType";
+        String hashTableTypeProperty = SystemPropertyUtil.get(hashTableTypeKey, "newarray");
+
+        hashTableTypeProperty = hashTableTypeProperty.toLowerCase(Locale.US).trim();
+
+        HashType hashTableType;
+        if ("newarray".equals(hashTableTypeProperty)) {
+            hashTableType = HashType.NEW_ARRAY;
+            logger.debug("-D{}=\"{}\"", hashTableTypeKey, hashTableTypeProperty);
+        } else if ("reuse".equals(hashTableTypeProperty)) {
+            hashTableType = HashType.FAST_THREAD_LOCAL_ARRAY_FILL;
+            logger.debug("-D{}=\"{}\"", hashTableTypeKey, hashTableTypeProperty);
+        } else {
+            hashTableType = HashType.NEW_ARRAY;
+            logger.debug("-D{}=\"newarray\" (unknown: {})", hashTableTypeKey, hashTableTypeProperty);
+        }
+
+        DEFAULT_HASH_TABLE_TYPE = hashTableType;
+    }
+
+    public Snappy() {
+        this(DEFAULT_HASH_TABLE_TYPE);
+    }
+
+    Snappy(HashType hashTableType) {
+        this.hashTableType = hashTableType;
+    }
+
+    public static Snappy withHashTableReuse() {
+        return new Snappy(HashType.FAST_THREAD_LOCAL_ARRAY_FILL);
+    }
+
+    private final HashType hashTableType;
     private State state = State.READING_PREAMBLE;
     private byte tag;
     private int written;
 
+    public enum HashType {
```

Comment:
this should be private
```suggestion
   private enum HashType {
```

Reply:
you can drop using an enum and just a boolean property based on the configured sys property: it would work better because will make it a proper constant, for the JIT (see https://shipilev.net/jvm/anatomy-quarks/17-trust-nonstatic-final-fields/ for more info)